tests/karate/src/test/java/common/net/MultiMsgWebSocketClient.java

```diff
@@ -41,113 +42,114 @@ public MultiMsgWebSocketClient(WebSocketOptions options, Logger logger, MessageQ
     setTextHandler(m -> true);
   }
 
-  public void send(Map<String, Object> jsonDataMap){
-    this.send(Json.of(jsonDataMap).toString());
-  }
-
-  @Override
-  public void signal(Object result) {
-    logger.trace("signal called: {}", result);
-    queue.onNewMsg(result.toString());
-  }
-
-  @Override
-  public synchronized Object listen(long timeout) {
-    logger.trace("entered listen wait state");
-    String msg = queue.take();
-
-    if (msg == null) logger.error("listen timed out");
-
-    return msg;
-  }
-
-  public MessageBuffer getBuffer() {
-    return queue;
+  /**
+   * JSON messages defined inside features are interpreted as maps from String to Object.
+   * This method is called directly inside features and is just a wrapper around the send method of WebsocketClient (that takes Strings).
+   * @param messageData the message to send as a JSON map
+   *                    (for example: subscribe and catchup in simpleScenarios.feature).
+   */
+  public void send(Map<String, Object> messageData){
+    this.send(mapToJsonString(messageData));
```

Comment:
I'd consider renaming `messageData` to something like `wsMessage` or `jsonRpcMsg` to indicate that we're talking about the lowest level.
`messageData` is already pretty (over)loaded as a concept in this project.